core/adapter/adapter_registry.py

```
@@ -454,13 +463,27 @@ async def register_adapter(self, info: AdapterInfo):
+        exc = task.exception()
+        if exc:
+            logger.error(f"Adapter {name} task failed: {exc}", exc_info=exc)
+        else:
+            logger.info(f"Adapter {name} task finished")
+
     async def start_adapter(self, name):
         """start an adapter by specified adapter name"""
         try:
             task = asyncio.create_task(self._adapters[name].start())
-            task.add_done_callback(lambda t: logger.info(f"Started adapter {name}"))
         except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)